src/ts/ssh/progress.ts

```diff
@@ -0,0 +1,163 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//
+
+import { AuthenticationMessage } from "./messages/authenticationMessages";
+import {
+	ChannelCloseMessage,
+	ChannelFailureMessage,
+	ChannelOpenConfirmationMessage,
+	ChannelOpenFailureMessage,
+	ChannelOpenMessage,
+	ChannelSuccessMessage } from "./messages/connectionMessages";
+import {
+	KeyExchangeDhInitMessage,
+	KeyExchangeDhReplyMessage,
+	NewKeysMessage } from "./messages/kexMessages";
+import { SshMessage } from "./messages/sshMessage";
+import {
+	ExtensionInfoMessage,
+	ServiceAcceptMessage,
+	ServiceRequestMessage,
+	SessionRequestSuccessMessage } from "./messages/transportMessages";
+
+/**
+ * Connection progress
+ */
+export enum Progress {
```

Comment:
The version exchange is one of the most important steps to report on because it validates the 2 sides are communicating at all. It does not use the standard message protocol so it needs to be handled as a special case.